[LP-47941; LP-47943; LP-47944; LP-47948] Feat: Adds datasets details

l-mansouri opened the pull request:
## Overview

This PR does the following:

- Adds command to list details of files `cloudos datasets ls <path> --profile <profile> --details`

## Jira

[LP-47941](https://lifebit.atlassian.net/browse/LP-47941)
[LP-47943](https://lifebit.atlassian.net/browse/LP-47943)
[LP-47944](https://lifebit.atlassian.net/browse/LP-47944)
[LP-47948](https://lifebit.atlassian.net/browse/LP-47948)

## `cloudos datasets ls --help`

![Screenshot 2025-06-10 at 12 07 11](https://github.com/user-attachments/assets/77c61830-e321-4072-a20a-8c80c8a96168)

## Tests

First, configure your profile using `cloudos configure --profile <name>`. 


> **Note:** The following examples are run in internal prod cloudos, with project `leila-testing`. 

### `cloudos datasets ls --profile <my_profile> --details`
![Screenshot 2025-06-10 at 12 29 36](https://github.com/user-attachments/assets/37773448-593c-4c43-b476-bf7efe584ba2)

### `cloudos datasets ls --profile <my_profile> --details Data` - this is the same as using `AnalysesResults` or `Cohort`
![Screenshot 2025-06-10 at 12 30 53](https://github.com/user-attachments/assets/6c6ddc05-6b6a-41bb-8d3c-54f2860bff7a)

### `cloudos datasets ls --profile <my_profile> --details Data/test_folder` - test_folder is a virtual folder

![Screenshot 2025-06-10 at 12 32 50](https://github.com/user-attachments/assets/044f2449-061d-4411-96e1-b14e1967fbee)

### `cloudos datasets ls --profile <my_profile> --details Data/test_folder/testing` - testing is another virtual folder (nested virtual folders to show the virtual folder recursion)

![Screenshot 2025-06-10 at 12 34 26](https://github.com/user-attachments/assets/70703259-4c6e-4fdd-83a0-d87f98fc9e68)

### `cloudos datasets ls --profile <my_profile> --details Data/test_folder/testing/input_data` - input data is a s3 folder
![Screenshot 2025-06-10 at 12 35 28](https://github.com/user-attachments/assets/0a47e074-9b94-436f-914e-0ae75f44135a)

### `cloudos datasets ls --profile  <my_profile> --details AnalysesResults/MR_1st_test-6791ff437f7769e3f9e094f4/results/twosampleMR_inputs` - nested s3 folders

![Screenshot 2025-06-10 at 12 37 03](https://github.com/user-attachments/assets/848e3085-1a4e-400a-8f05-d4f0ab787c7f)

dapineyro commented on cloudos_cli/__main__.py:
```diff
@@ -1832,6 +1834,19 @@ def run_bash_job(ctx,
               f'\t\t--job-id {j_id}\n')
 
 
+
+def format_bytes(size):
+    """Convert bytes to human-readable format (e.g., 1.2 MB)."""
+    if size is None:
+        return "-"
+    power = 1024
+    n = 0
+    labels = ['B', 'KB', 'MB', 'GB', 'TB', 'PB']
+    while size >= power and n < len(labels) - 1:
+        size /= power
+        n += 1
+    return f"{size:.1f} {labels[n]}"
+
```
Could we move this function (alongside with `ssl_selector` ) into a new script in `cloudos-cli/utils/resources.py` or similar. Wdyt? It will be better organised, right?

dapineyro commented on cloudos_cli/__main__.py:
```diff
@@ -1914,15 +1935,79 @@ def list_files(ctx,
+                click.echo(name)
     except Exception as e:
         click.echo(f"[ERROR] {str(e)}", err=True)
 
```
```suggestion
```
Remove last blank line please

dapineyro commented on cloudos_cli/datasets/datasets.py:
```diff
@@ -150,7 +150,13 @@ def list_project_content(self):
+
+        #  Normalize response
+        for item in raw.get("datasets", []):
+            item ["folderType"]=True
```
```suggestion
            item["folderType"] = True
```

l-mansouri replied:
I was trying to do it from VS code and it messed up a lot of things. I have learned my lesson, sorry!

l-mansouri commented:
I have addressed all the comments.
The table now looks like this
![Screenshot 2025-06-11 at 10 13 13](https://github.com/user-attachments/assets/4a0b57c5-70b5-4307-ae74-95c9fb4e05d5)

dapineyro commented on CHANGELOG.md:
```diff
@@ -1,10 +1,9 @@
+### Feat
 
-- Fix PyPi build
+- Adds command to list details of files `cloudos datasets ls <path> --profile <profile> --details`
```
You have removed my previous patch v2.26.1, could you please add it back?

dapineyro commented on cloudos_cli/__main__.py:
```diff
@@ -8,11 +8,14 @@
 from ._version import __version__
 from cloudos_cli.configure.configure import ConfigurationProfile
 from cloudos_cli.datasets import Datasets
+import re
```
```suggestion
```
Not used

dapineyro commented on cloudos_cli/__main__.py:
```diff
@@ -8,11 +8,14 @@
+from cloudos_cli.utils.resources import ssl_selector, format_bytes
+from rich.console import Console
+from rich.table import Table
+from rich.style import Style
 
```
```suggestion
from rich.style import Style


```
Expected 2 blank lines

dapineyro reviewed (changes requested):
Hey @l-mansouri I have some more comments. Also I completely agree with you that the cropped s3 path won't be really useful. Could we come back to the original implementation? Sorry...

dapineyro reviewed (approved):
Now it looks amazing! Thanks @l-mansouri
